### PR TITLE
fix: UAT3 Testing Feedback

### PR DESCRIPTION
**Pull request by rutalreja-deloitte**
Fixes

- Repo Folder structure
- Readmes to fix linking on folder names
- gitignore files
- API Add to GKE Project - serviceusage.googleapis.com
- Bastion don't create fw flag
- Bastion SA to get Source Writer in app-cicd proj
- Bin Auth add to whitelist
- Enforce Bin Auth off on dev and non prod - default tfvar change
- FW Rule Allow egress all ips all protocol 2022 port
- Node Tag egress-internet
- Ingress FW Rules for cross node ingress used hardcoded dev cidrs changed to vars coming from specific envs

### CODE REVIEW

**Comment by rutalreja-deloitte** [comment]
@bharathkkb should be ready for review

File: 5-infrastructure/modules/base-env/boa-gke.tf [hunk 1]
```diff
@@ -58,7 +58,7 @@ locals {
     }
   }
   bin_auth_attestors = [for attestor in var.bin_auth_attestor_names : "projects/${var.bin_auth_attestor_project_id}/attestors/${attestor}"]
-  allowlist_patterns = ["quay.io/random-containers/*", "k8s.gcr.io/more-random/*", "gcr.io/${var.boa_gke_project_id}/*"] # Example
+  allowlist_patterns = ["quay.io/random-containers/*", "k8s.gcr.io/more-random/*", "gcr.io/${var.boa_gke_project_id}/*", "gcr.io/config-management-release/*"] # Example
```
Comment (bharathkkb) [inline 5-infrastructure/modules/base-env/boa-gke.tf:61]: @rutalreja-deloitte @dkassab  could you check if ASM pods started?

File: 5-infrastructure/modules/bastion/main.tf [hunk 2]
```diff
@@ -51,7 +51,14 @@ module "iap_bastion" {
...
-  shielded_vm = false
-  members     = var.bastion_members
-  tags        = ["bastion", "allow-google-apis", "egress-internet"]
+  create_firewall_rule = false
+  shielded_vm          = false
+  members              = var.bastion_members
+  tags                 = ["bastion", "allow-google-apis", "egress-internet"]
+}
+
+resource "google_project_iam_member" "app_cicd_cloudbuild_sa_roles" {
```
Comment (bharathkkb) [inline 5-infrastructure/modules/bastion/main.tf:60]:
```suggestion
resource "google_project_iam_member" "bastion_repo_access" {
```

File: README.md [hunk 3]
```diff
@@ -166,7 +166,7 @@ The purpose of this step is to install the Anthos components required for the Ba
 - All required components to install **Anthos Config Management**
 - All required components to install **Anthos Service Mesh**
 
-## [7. build_app](./build_app)
+## [7. build_app](./7-app-build-deploy)
```
Comment (bharathkkb) [inline README.md:169]:
```suggestion
## [7. app-build-deploy](./7-app-build-deploy)
```

File: 5-infrastructure/modules/base-env/boa-gke.tf [hunk 4]
```diff
@@ -116,7 +117,7 @@ module "clusters" {
     "mesh_id" = "proj-${data.google_project.gke_project.number}"
   }
   node_pools_tags = {
-    "np-${each.value.region}" : ["boa-${each.key}-cluster", "allow-google-apis"]
+    "np-${each.value.region}" : ["boa-${each.key}-cluster", "allow-google-apis", "egress-internet"]
```
Comment (bharathkkb) [inline 5-infrastructure/modules/base-env/boa-gke.tf:120]: lets add a common `boa-cluster` tag across all clusters, this way we can target via single tag in `target_tags`

File: foundation-extension/3-networks-extension/modules/fw-rules/main.tf [hunk 5]
```diff
@@ -65,6 +65,24 @@ locals {
...
+      destination_ranges = ["0.0.0.0/0"]
+      target_tags        = ["boa-gke2-cluster"]
+      allow_protocol     = "all"
+      allow_ports        = ["2022"]
+    },
+    "fw-${var.environment_code}-shared-base-e-mci-2022" = {
+      destination_ranges = ["0.0.0.0/0"]
+      target_tags        = ["boa-mci-cluster"]
+      allow_protocol     = "all"
+      allow_ports        = ["2022"]
```
Comment (bharathkkb) [inline foundation-extension/3-networks-extension/modules/fw-rules/main.tf:85]: Can you add a comment here for why we have this and link to an issue, so we can cleanup in the future

File: foundation-extension/3-networks-extension/modules/fw-rules/main.tf [hunk 6]
```diff
@@ -65,6 +65,24 @@ locals {
...
+    "fw-${var.environment_code}-shared-base-e-gke2-2022" = {
+      destination_ranges = ["0.0.0.0/0"]
+      target_tags        = ["boa-gke2-cluster"]
+      allow_protocol     = "all"
+      allow_ports        = ["2022"]
+    },
+    "fw-${var.environment_code}-shared-base-e-mci-2022" = {
+      destination_ranges = ["0.0.0.0/0"]
+      target_tags        = ["boa-mci-cluster"]
+      allow_protocol     = "all"
```
Comment (bharathkkb) [inline foundation-extension/3-networks-extension/modules/fw-rules/main.tf:84]: lets do just tcp

**Review by bharathkkb — CHANGES_REQUESTED** [review]

File: 5-infrastructure/modules/base-env/boa-gke.tf (hunk 4 above)
Comment (rutalreja-deloitte) [reply 5-infrastructure/modules/base-env/boa-gke.tf:120]: the seperate tags allow me to target one specific cluster in fw rules, but I can add a 4th tag "boa-cluster"

**Review by daniel-cit — COMMENTED** [review]
It is also necessary to replace `build_app` with `7-app-build-deploy` in the file
https://github.com/rutalreja-deloitte/terraform-example-foundation-app/blob/uat3/7-app-build-deploy/README.md

**Comment by rutalreja-deloitte** [comment]
New Fix after first review:
Ingress FW Rules for cross node ingress used hardcoded dev cidrs changed to vars coming from specific envs

File: 5-infrastructure/modules/base-env/boa-gke.tf (hunk 4 above)
Comment (bharathkkb) [reply 5-infrastructure/modules/base-env/boa-gke.tf]: sorry I meant in addition to existing as well, that sgtm

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**